context/FormProvider.tsx

```diff
@@ -0,0 +1,158 @@
+import { useAccount, useContractRead } from "@starknet-react/core";
+import React, { FunctionComponent, useState } from "react";
+import { createContext, useMemo } from "react";
+import { computeMetadataHash, generateSalt } from "@/utils/userDataService";
+import { useNftPpVerifierContract } from "@/hooks/contracts";
+import { Abi } from "starknet";
+import { filterAssets, retrieveAssets } from "@/utils/nftService";
+
+interface FormState {
+  email: string;
+  isSwissResident: boolean;
+  tokenId: number;
+  duration: number;
+  selectedDomains: Record<string, boolean>;
+  // metadata
+  salt?: string;
+  metadataHash?: string;
+  needMetadata: boolean;
+  // tax
+  salesTaxRate: number;
+  //pfp
+  selectedPfp?: StarkscanNftProps;
+  //upsell
+  isUpselled: boolean;
+}
+
+type FormConfig = {
+  formState: FormState;
+  clearForm: () => void;
+  updateFormState: (updates: Partial<FormState>) => void;
+  userNft?: StarkscanNftProps[];
+};
+
+const initialState: FormState = {
+  email: "",
+  isSwissResident: false,
+  tokenId: 0,
+  duration: 1,
+  selectedDomains: {},
+  needMetadata: false,
+  salesTaxRate: 0,
+  selectedPfp: undefined,
+  isUpselled: false,
+};
+
+export const FormContext = createContext<FormConfig>({
+  formState: initialState,
+  clearForm: () => {},
+  updateFormState: () => {},
+  userNft: [],
```

Comment:
why do you need to initialize everything with values like this? Could you just leave this undefined or ?

Reply:
you mean not defining `formState` and `userNft` ?